proposals/nnnn-haptic-projection.md

```diff
@@ -0,0 +1,124 @@
+
+Examples of required changes:
+
+```java
+class RPCRequestFactory {
+    public static SendHapticData buildSendHapticData(List<HapticRect> hapticRectData, 
+            Integer correlationID) {
+        SendHapticData msg = new SendHapticData();
+        msg.setCorrelationID(correlationID);
+        msg.setHapticRectData(hapticRectData);
+        return msg;
+    }
+}
+```
+```java
+class SdlProxyBase {
+    	public void sendHapticData(List<HapticRect> hapticRectData) throws SdlException {
+    	    final SendHapticData msg = RPCRequestFactory.buildSendHapticData(hapticRectData, 200);
+    	    sendRPCRequest(msg);
+    	}
+}
+```
+```java
+class VirtualDisplayEncoder {
+    class SdlPresentation extends Presentation {
+        
+        private List<HapticRect> userSpecifiedHaptics;
+        
+        @Override
+        protected void onCreate(Bundle savedInstanceState) {
+            ...
+            setOnShowListener(new OnShowListener() {
+               @Override
+               public void onShow(DialogInterface dialog) {
+                   // If the developer manually sets the haptics, do not automatically calculate
+                   if (userSpecifiedHaptics != null)
+                       refreshHapticData();
+               } 
+            });
+        }
+        
+        /**
+         * Automatically figure out the HapticRects for this Presentation
+         */
+        public final void refreshHapticData() {
+            List<HapticRect> list = new ArrayList<>();
+            findHapticRects(w.getDecorView(), list);
+            try {
+                proxy.sendHapticData(list);
+            } catch (SdlException e) {
+                e.printStackTrace();
+            }
+        }
+        
+        /**
+         * Walks the view hierarchy to find focusable views 
+         */
+        private static void findHapticRects(View view, final List<HapticRect> list) {
+            if (view.isFocusable()) {
+                // Create HapticRect and add it to the list
+            }
+            if (view instanceof ViewGroup) {
+                // Recursively check children 
+            }
+        }
+        
+        /**
+         * The developer can manually call this to override the default view traversal
+         */
+        public final void setHapticData(List<HapticRect> hapticData) {
+            userSpecifiedHaptics = hapticData;
+            try {
+                proxy.sendHapticData(hapticData);
+            } catch (SdlException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+} 
+```
+## Potential downsides
+
+1. This does not monitor changes in the view hierarchy (i.e. adding/removing a view, visibility 
+changes nor enable/disable). Any runtime changes will need to be manually calculated by the 
+developer or `SdlPresentation#refreshHapticData()` will need to be invoked again.
+2. This does not take into consideration layout overrides described [here](https://developer
```

Comment:
There's a space after "developer" that's causing this link to break.  Please remove the space so that the link works properly.